src/Base58.cs

```diff
@@ -284,24 +279,28 @@ public bool TryDecode(ReadOnlySpan<char> input, Span<byte> output, out int numBy
         }
 
         int zeroCount = getPrefixCount(input, ZeroChar);
-        return internalDecode(
+        bool result = internalDecode(
             input,
             output,
             zeroCount,
-            out numBytesWritten);
+            out int bytesWrittenRangeStart,
+            out int bytesWrittenRangeEnd);
+
+        output[bytesWrittenRangeStart..bytesWrittenRangeEnd].CopyTo(output);
+        numBytesWritten = bytesWrittenRangeEnd - bytesWrittenRangeStart;
+        return result;
     }
 
     static void computeDoubleSha256(ReadOnlySpan<byte> buffer, Span<byte> output)
     {
         Span<byte> tempResult = stackalloc byte[sha256Bytes];
-        using var sha256 = SHA256.Create();
```

Comment:
I think the reason I kept a single instance of `SHA256` was that I assumed this was using the continuation of the state machine for the second SHA256 operation. Obviously, it isn't the case if the tests pass.